catalog/app/components/PaymentDialog/index.js

```diff
@@ -28,8 +27,14 @@ const style = {
 };
 
 class PaymentDialog extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
-  state = {
-    showConfirm: false,
+  handleConfirm(curr, next, onDowngrade) {
+    if (next.confirmTitle) {
+      // eslint-disable-next-line no-alert no-restricted-globals
+      const proceed = window.confirm(`${next.confirmTitle}\n${next.confirmBody}`);
```

Comment:
`window.confirm`???

![oh no you didn't](http://78.media.tumblr.com/907a1d401de4d7ac7ff616176449917d/tumblr_mqtmqa6yMs1s2gerco1_500.gif)

Reply:
lol. i actually ripped out a pretty custom dialog to do this. i think it's clearer (scarier) and it is definitely better on mobile.